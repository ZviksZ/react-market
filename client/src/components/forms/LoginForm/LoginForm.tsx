import * as React      from 'react'
import { useForm }     from 'react-hook-form'
import { loginSchema } from '../../../services/helpers/validation'
import { useDispatch } from 'react-redux'
import { TextField }   from '@material-ui/core'
import Button          from '@material-ui/core/Button'
import { yupResolver } from '@hookform/resolvers/yup'
import Grid            from '@material-ui/core/Grid'

interface IFormInputs {
	login: string
	password: string
}

export const LoginForm: React.FC = () => {
	const dispatch = useDispatch()

	const { register, handleSubmit, errors } = useForm<IFormInputs>({
		resolver: yupResolver(loginSchema),
	})

	const onSubmit = (data: any) => {
		console.log(data)
		//dispatch(login(data.login, data.password))
	}

	return (
		<form onSubmit={handleSubmit(onSubmit)}>
			<Grid container spacing={2}>
				<Grid item xs={12}>
					<TextField
						variant="outlined"
						margin="normal"
						required
						fullWidth
						id="username"
						label="Username"
						error={!!errors.login}
						helperText={errors.login ? errors.login.message : ''}
						inputRef={register}
						name="username"
						autoComplete="email"
						autoFocus
						className="form-input"
					/>
				</Grid>
				<Grid item xs={12}>
					<TextField
						variant="outlined"
						margin="normal"
						required
						fullWidth
						name="password"
						error={!!errors.password}
						helperText={errors.password ? errors.password.message : ''}
						inputRef={register}
						label="Password"
						type="password"
						id="password"
						className="form-input"
						autoComplete="current-password"
					/>
				</Grid>
				<Grid item xs={12}>
					<Button type="submit" color="primary" size="large" fullWidth variant="contained">
						Login
					</Button>
				</Grid>
			</Grid>
		</form>
	)
}
