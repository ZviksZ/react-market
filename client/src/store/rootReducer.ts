import { combineReducers } from 'redux'
import { cartReducer } from './ducks/cart/reducer'
import { productsReducer } from './ducks/product/reducer'
import { authReducer } from './ducks/auth/reducer'

export const rootReducer = combineReducers({
	products: productsReducer,
	cart: cartReducer,
	auth: authReducer,
})
