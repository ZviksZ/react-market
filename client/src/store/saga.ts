import { all } from 'redux-saga/effects'
import { productsSaga } from './ducks/product/sagas'

export default function* rootSaga() {
	yield all([productsSaga()])
}
