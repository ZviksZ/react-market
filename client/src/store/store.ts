import { applyMiddleware, compose, createStore } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { rootReducer } from './rootReducer'
import rootSaga from './saga'
import { ProductsState } from './ducks/product/contracts/state'
import { CartState } from './ducks/cart/contracts/state'
import { AuthState } from './ducks/auth/contracts/state'

const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose

const sagaMiddleware = createSagaMiddleware()

export interface RootState {
	products: ProductsState
	cart: CartState
	auth: AuthState
}

export const store = createStore(rootReducer, composeEnhancers(applyMiddleware(sagaMiddleware)))

sagaMiddleware.run(rootSaga)
