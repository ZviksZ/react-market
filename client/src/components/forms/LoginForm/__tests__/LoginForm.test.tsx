import { cleanup, fireEvent, render } from '@testing-library/react'
import * as React                     from 'react'
import { shallow }                    from 'enzyme'
import { LoginForm }                  from '../LoginForm'
import { Provider }                   from 'react-redux'
import { store }                      from '../../../../store/store'

describe('LoginForm testing', () => {
	afterEach(cleanup)

	it('take form snapshot', () => {
		const form = shallow(
			<Provider store={store}>
				<LoginForm />
			</Provider>,
		)

		expect(form).toMatchSnapshot()
	})
})
