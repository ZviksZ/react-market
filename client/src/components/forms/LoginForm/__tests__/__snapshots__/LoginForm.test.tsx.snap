// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm testing take form snapshot 1`] = `ShallowWrapper {}`;
